File: packages/sdk-ts/src/clients/triggers.ts
import type { HttpClient } from '../http.js';
import type { CreateTriggerParams, Trigger } from '../types.js';

export class TriggerClient {
  constructor(private readonly http: HttpClient) {}

  /** Create a memory trigger. */
  async create(params: CreateTriggerParams): Promise<Trigger> {
    return this.http.request('POST', '/v1/triggers', params);
  }

  /** List triggers for a scope. */
  async list(scope: string, scopeId: string): Promise<Trigger[]> {
    return this.http.request('GET', `/v1/triggers?scope=${scope}&scope_id=${scopeId}`);
  }

  /** Delete a trigger. */
  async delete(triggerId: string): Promise<void> {
    return this.http.request('DELETE', `/v1/triggers/${triggerId}`);
  }
}

File: packages/sdk-python/steno_ai/triggers.py
"""Trigger client — create and manage memory triggers."""

from typing import Dict, List

from .client import HttpClient


class TriggerClient:
    def __init__(self, http: HttpClient):
        self._http = http

    def create(self, scope: str, scope_id: str, condition: dict) -> dict:
        """Create a memory trigger."""
        return self._http.request("POST", "/v1/triggers", json={
            "scope": scope,
            "scope_id": scope_id,
            "condition": condition,
        })

    def list(self, scope: str, scope_id: str) -> list:
        """List triggers for a scope."""
        return self._http.request("GET", f"/v1/triggers?scope={scope}&scope_id={scope_id}")

    def delete(self, trigger_id: str) -> None:
        """Delete a trigger."""
        self._http.request("DELETE", f"/v1/triggers/{trigger_id}")
